README.md

```diff
@@ -1,2 +1,300 @@
+
+### ``CuriePower.doze(int duration)``
+
+```
+void CuriePower.doze(int duration)
+```
+
+
+Places the SoC in "doze" mode which switches the system clock to the internal 32.768 kHz RTC oscillator for `duration` milliseconds
+
+
+*Parameters*
+
+1. `int duration` : time in milliseconds to place the SoC in "doze" mode
+none
+
+*Return value*
+
+none
+
+### ``CuriePower.idle()``
+
+```
+void CuriePower.idle()
+```
+
+Places the SoC into "doze" mode then enters an infinite loop, effectively stopping all operations until a wake interrupt is generated.
+
+*Parameters*
+
+none
+
+*Return value*
+
+none
+
+### ``CuriePower.idle(int duration)``
+
+```
+void CuriePower.idle(int duration)
+```
+
+
+Places the SoC in "doze" mode and enters an infinite loop for `duration` milliseconds
+
+
+*Parameters*
+
+1. `int duration` : time in milliseconds to place the SoC in an idle state
+none
+
+*Return value*
+
+none
+
+### ``CuriePower.sleep()``
+
+```
+void CuriePower.sleep()
+```
+
+Places the SoC into a sleep state, stopping all operations, until a wake interrupt is generated
+
+*Parameters*
+
+none
+
+*Return value*
+
+none
+
+### ``CuriePower.sleep(int duration)``
+
+```
+void CuriePower.sleep(int duration)
+```
+
+
+Places the SoC into a sleep state for `duration` milliseconds
+
+
+*Parameters*
+
+1. `int duration` : time in milliseconds to place the SoC into a sleep state
+none
+
+*Return value*
+
+none
+
+### ``CuriePower.deepSleep()``
+
+```
+void CuriePower.deepSleep()
+```
+
+Places the SoC into a deep sleep state, stopping all operations, until a wake interrupt is generated
+
+*Parameters*
+
+none
+
+*Return value*
+
+none
+
+### ``CuriePower.deepSleep(int duration)``
+
+```
+void CuriePower.deepSleep(int duration)
+```
+
+
+Places the SoC into a deep sleep state for `duration` milliseconds
+
+
+*Parameters*
+
+1. `int duration` : time in milliseconds to place the SoC into a deep sleep state
+none
+
+*Return value*
+
+none
+
+### ``CuriePower.attachInterruptWakeup(uint32_t pin, voidFuncPtr callback, uint32_t mode)``
+
+```
+void CuriePower.attachInterruptWakeup(uint32_t pin, voidFuncPtr callback, uint32_t mode)
+```
+
+Attaches a wakeup interrupt to digital pin `pin`. `callback` callback is the function to be called when the wakeup interrupt occurs. 
+
+*Parameters*
+
+none
+
+*Return value*
+
+none
+
+### ``CuriePower.dettachInterruptWakeup(uint32_t pin)``
+
+```
+void CuriePower.dettachInterruptWakeup(uint32_t pin)
+```
+
+Removes any wakeup interrupt attached to digital pin `pin`. 
+
+*Parameters*
+
+none
+
+*Return value*
+
+none
+
+# Tutorials
+
+## Tutorial #1: TimedWakeup Example
+
+This sketch demonstrates the simplest way to use the CuriePower library. It blinks the LED a few times, goes to sleep for a certain amount then goes back at the start of loop()
+
+```cpp
+#include <Power.h>
+
+void setup() {
+  pinMode(LED_BUILTIN, OUTPUT);
+}
+
+void loop() {
+  for(int i = 0; i < 5; i++)
+  { 
+    digitalWrite(LED_BUILTIN, HIGH);
+    delay(100);
```

Comment:
Wouldn't you want to use PM.sleep() here as well? i.e. replace *all* delays with PM calls?